apps/src/p5lab/drawUtils.js

```diff
@@ -47,7 +47,9 @@ export function speechBubble(
   y,
   width,
   height,
-  {triangleSize = 10, fill = 'white', strokeWeight = 2, stroke = 'gray'} = {}
+  triangleSize,
+  tipX,
+  {fill = 'white', strokeWeight = 2, stroke = 'gray'} = {}
```

Comment:
`triangleSize` and `tipX` should be optional with default values, so they should go in this config object. the default for `tipX` should be x, which can be set in the same way the other defaults here are set even though it's also a parameter for this function:

```javascript
function speechBubble(x, y, {tipX = x} = {}) {}
```

tipX will default to x, and the config object will default to an empty object

Reply:
I'm finding that it seems to prefer any default values given inside the config object to values that are passed as additional parameters. In other words, if I set triangleSize to 10 inside this object, it's always 10, even if I call the function with a different value being passed.

Reply:
i think that's because you need to change the caller -- if you update the configuration object in the function signature but don't update where we're calling `drawUtils.speechBubble`, that's the outcome i'd expect. if you've already tried that, we can pair on it, but going a little deeper into what i mean:

```javascript
// drawUtils.js
function speechBubble(x, y, width, height, {triangleSize = 10, triangleTipX = x} = {}) {}

// CoreLibrary.js
// what we have now:
drawUtils.speechBubble(
  x,
  y,
  width,
  height,
  triangleSize,  // speechBubble thinks this is the config object
  triangleTipX  // speechBubble ignores this
);

// what we need to update to:
drawUtils.speechBubble(
  x,
  y,
  width,
  height,
  {triangleSize, triangleTipX}  // config object overrides
);
```

it's kind of hard to get used to, but we're using [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) as part of the `speechBubble` function signature, but it's the same as:

```javascript
function speechBubble(x, y, width, height, config) {
  const triangleSize = config.triangleSize;
}
```

Reply:
Oh got it! I can do this now.  Could you explain the rationale for which values go into the config object and which are just normal parameters?

Reply:
yeah, of course. especially for drawing functions, there can be _a ton_ of parameters that callers would want to have control over, so my general approach is: any info we _need_ from the caller because we can't draw the thing otherwise should be its own parameter, and anything else that we can set reasonable defaults for should be in a config object